### PR TITLE
Cleanup YAML formatting

### LINKED ISSUE #1313: YAML formatting should be cleaned up to linter standards
**What steps did you take and what happened:**

Run `make -C images/capi lint` and see that there are numerous warnings from `ansible-lint` that the project is ignoring.

In particular, there are many formatting warnings. Many of these boil down to what the linter considers basic YAML linting (removing unnecessary quotes, ensuring document separators `---`, and so on). It would be helpful to fix just those separately.

**What did you expect to happen:**

Running `make -C images/capi lint` shouldn't result in any basic formatting warnings.

**Anything else you would like to add:**

One way to accomplish this:

```shell
ansible-lint --fix=none images/capi/ansible
make -C images/capi lint-ignore
git add images/capi
git commit -m "Format YAML files for ansible-lint"
```

These commands:
- Tell the linter not to fix a specific type of warning, so it just does its basic formatting
- Regenerate the `.ansible-lint-ignore` file so these warnings won't be ignored in the future
- Add all the changed files to version control and commit them.

Then run `make -C images/capi lint` to ensure it's passing, and create a PR.

**Environment:**

N/A

/kind bug
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by FaithKovi**
What this PR does / why we need it:

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #1313

**Additional context**
Add any other context for the reviewers

### CODE REVIEW

**Comment by mboersma** [comment]
Thanks @FaithKovi! This looks much better.

Apparently there's a conflict with #1324 that just merged. Could you rebase the PR from the current `main` branch and force-push it? Sorry for the hassle!

**Comment by mboersma** [comment]
/ok-to-test

@FaithKovi I rebased this off `main`–I hope that's ok with you. I can take a look at fixing up the newlines today as well, and hopefully it's ready to merge after that.

File: images/capi/ansible/windows/roles/cloudbase-init/tasks/main.yml [hunk 1]
```diff
@@ -15,33 +15,33 @@
 - name: Download Cloudbase-init
   win_get_url:
     url: "{{ cloudbase_init_url }}"
-    dest: '{{ tempdir.stdout | trim }}\CloudbaseInitSetup.msi'
+    dest: "{{ tempdir.stdout | trim }}\\CloudbaseInitSetup.msi"
```
Comment (AverageMarcus) [inline images/capi/ansible/windows/roles/cloudbase-init/tasks/main.yml:18]: Just to double check, are these backslashes supposed to be escaped? I would have expected them to also need to be with the quotes.

**Review by AverageMarcus — COMMENTED** [review]
/lgtm

Just want to double check those backslashes are correct.

**Comment by AverageMarcus** [comment]
Thank you so much for being so patient with this @FaithKovi

**Review by drew-viles — APPROVED** [review]
LGTM Thanks @FaithKovi !!

**Pull request merged**